### ISSUE TITLE
Need more flexible way to handle language tags

### ISSUE THREAD

**Issue by NorbertLindenberg** [MicrosoftDocs/typography-issues#275]
OpenType identifies languages with its own set of language tags, while applications that support language identification (such as web browsers) tend to use BCP 47 language tags, which combine ISO 639 language tags, ISO 15924 script tags, ISO 3166-1 country codes, and possibly more information.

Getting support for a language into OpenType, with proper mapping from BCP 47 language tags to OpenType language tags, is a long process: Developers ask Microsoft to register OpenType language tags for all the languages they care about, the registry gets updated and published, font developers can use the new OpenType language tags in their fonts, and over the course of the next decade, as OpenType rendering system implementers add the new language tags, and updated implementations get into OS and app updates, and users update their software or replace their Android phones, more and more users get to see text rendered correctly.

HarfBuzz has a partial workaround: Any three-letter code that isn’t already registered as an OpenType language tag is uppercased and treated as an OpenType language tag, so that any ISO 639 3-letter language code is covered one way or another (hb_ot_tags_from_language). This works as long as all the languages needed in a font have only 3-letter codes.

It would be good to a standard solution to the problem in OpenType, such as:
* Standardize the workaround that's currently in HarfBuzz.
* Add a way for fonts to define the mapping from BCP 47 language tag to OpenType language tag in a table similar to the ltag table in Apple Advanced Typography.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 662c6f54-f90c-52ed-dca2-0f0a0a49fee5
* Version Independent ID: 6aa33924-d45e-60c3-a9fd-e317ff662993
* Content: [Language system tags - Typography](https://docs.microsoft.com/en-us/typography/opentype/spec/languagetags)
* Content Source: [typographydocs/opentype/spec/languagetags.md](https://github.com/MicrosoftDocs/typography/blob/live/typographydocs/opentype/spec/languagetags.md)
* Product: **typography**

**Comment by PeterCon** [comment]
Agreed that something would be good—I've thought that for many years. I did propose something a number of years ago (maybe on the opentype list?), but IIRC I didn't get much support. (Not strong opposition; just not much feedback to drive progress.) Since then Martin Hosken also brought it up, but I forget now the context.

Maintaining lang sys tags was always a chore, making sure 639 mappings are clear (enough) and nothing contradicts. But I no longer have access to the Excel file in which I kept all the extra details. So, I can say I'd be in support of something that reduces maintenance cost.

**Comment by PeterCon** [comment]
Cf. #274
